Test Failure with 0.7.0

I am getting [this log](http://bpaste.net/show/24caf8b4738d) for building and testing pandas-datareader.
For your convenience find the fail snippet below (look up the log for more details regarding the testing environment):

```
=================================================== FAILURES ===================================================
_______________________________________ TestYahoo.test_yahoo_DataReader ________________________________________

self = <pandas_datareader.tests.yahoo.test_yahoo.TestYahoo object at 0x7f67376ce5d0>

    def test_yahoo_DataReader(self):
        start = datetime(2010, 1, 1)
        end = datetime(2015, 5, 9)
        result = web.DataReader('AAPL', 'yahoo-actions', start, end)
    
        exp_idx = pd.DatetimeIndex(['2015-05-07', '2015-02-05',
                                    '2014-11-06', '2014-08-07',
                                    '2014-06-09', '2014-05-08',
                                    '2014-02-06', '2013-11-06',
                                    '2013-08-08', '2013-05-09',
                                    '2013-02-07', '2012-11-07',
                                    '2012-08-09'])
    
        exp = pd.DataFrame({'action': ['DIVIDEND', 'DIVIDEND', 'DIVIDEND',
                                       'DIVIDEND', 'SPLIT', 'DIVIDEND',
                                       'DIVIDEND', 'DIVIDEND',
                                       'DIVIDEND', 'DIVIDEND', 'DIVIDEND',
                                       'DIVIDEND', 'DIVIDEND'],
                            'value': [0.52, 0.47, 0.47, 0.47, 0.14285714,
                                      3.29, 3.05, 3.05, 3.05,
                                      3.05, 2.65, 2.65, 2.65]},
                           index=exp_idx)
        exp.index.name = 'Date'
>       tm.assert_frame_equal(result.reindex_like(exp).round(2), exp.round(2))

pandas_datareader/tests/yahoo/test_yahoo.py:258: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python2.7/site-packages/pandas/util/testing.py:1348: in assert_frame_equal
    obj='DataFrame.iloc[:, {idx}]'.format(idx=i))
/usr/lib64/python2.7/site-packages/pandas/util/testing.py:1227: in assert_series_equal
    obj='{obj}'.format(obj=obj))
pandas/_libs/testing.pyx:59: in pandas._libs.testing.assert_almost_equal
    ???
pandas/_libs/testing.pyx:173: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'DataFrame.iloc[:, 1]', message = 'DataFrame.iloc[:, 1] values are different (61.53846 %)'
left = '[0.52, 0.47, 0.47, 0.47, 0.14, 23.03, 21.35, 21.35, 21.35, 21.35, 18.55, 18.55, 18.55]'
right = '[0.52, 0.47, 0.47, 0.47, 0.14, 3.29, 3.05, 3.05, 3.05, 3.05, 2.65, 2.65, 2.65]', diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        elif is_categorical_dtype(left):
            left = repr(left)
    
        if PY2 and isinstance(left, string_types):
            # left needs to be printable in native text type in python2
            left = left.encode('utf-8')
    
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
        elif is_categorical_dtype(right):
            right = repr(right)
    
        if PY2 and isinstance(right, string_types):
            # right needs to be printable in native text type in python2
            right = right.encode('utf-8')
    
        msg = """{obj} are different
    
    {message}
    [left]:  {left}
    [right]: {right}""".format(obj=obj, message=message, left=left, right=right)
    
        if diff is not None:
            msg += "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: DataFrame.iloc[:, 1] are different
E       
E       DataFrame.iloc[:, 1] values are different (61.53846 %)
E       [left]:  [0.52, 0.47, 0.47, 0.47, 0.14, 23.03, 21.35, 21.35, 21.35, 21.35, 18.55, 18.55, 18.55]
E       [right]: [0.52, 0.47, 0.47, 0.47, 0.14, 3.29, 3.05, 3.05, 3.05, 3.05, 2.65, 2.65, 2.65]

/usr/lib64/python2.7/site-packages/pandas/util/testing.py:1018: AssertionError
```

Is this a known failure, if not, am I doing something wrong?

--- a/docs/source/whatsnew/v0.8.0.txt
+++ b/docs/source/whatsnew/v0.8.0.txt
@@ -1,0 +1,31 @@
+.. _whatsnew_080:
+
+v0.8.0 (TBD)
+---------------------------
+
+Highlights include:
+
+.. contents:: What's new in v0.8.0
+    :local:
+    :backlinks: none
+
+.. _whatsnew_080.enhancements:
+
+Enhancements
+~~~~~~~~~~~~
+
+
+.. _whatsnew_080.api_breaking:
+
+Backwards incompatible API changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+
+.. _whatsnew_080.bug_fixes:
+
+Bug Fixes
+~~~~~~~~~
+
+- Fix Yahoo! actions issue where dividends are adjusted twice as a result of a
+  change to the Yahoo! API. (:issue: `583`)

--- a/pandas_datareader/yahoo/daily.py
+++ b/pandas_datareader/yahoo/daily.py
@@ -181,11 +181,10 @@ class YahooDailyReader(_DailyBaseReader):
                 splits['Splits'] = 1.0 / splits['SplitRatio']
                 prices = prices.join(splits['Splits'], how='outer')
 
-                if 'DIVIDEND' in types and not self.adjust_dividends:
+                if 'DIVIDEND' in types and self.adjust_dividends:
                     # Adjust dividends to deal with splits
                     adj = prices['Splits'].sort_index(ascending=False).fillna(
                         1).cumprod()
-                    adj = 1.0 / adj
                     prices['Dividends'] = prices['Dividends'] * adj
 
         return prices

